File: YLTextViewDemo/YLTextViewDemo/ViewController.m
//
//  ViewController.m
//  YLTextViewDemo
//
//  Created by 张雨露 on 2016/12/23.
//  Copyright © 2016年 张雨露. All rights reserved.
//

#import "ViewController.h"
#import "UITextView+YLTextView.h"
@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    self.view.backgroundColor = [UIColor lightGrayColor];
    
    UITextView *textView = [[UITextView alloc] init];
    textView.frame = CGRectMake(0, 0, 203, 120);
    textView.center = self.view.center;
    textView.font = [UIFont systemFontOfSize:17];
    
//    textView.text = @"我是一段很长的字符串，很长很长真的很长";
    textView.placeholder = @"我是一个占位符";
//    textView.limitLength = @10;
    textView.placeholdColor = [UIColor redColor];
    textView.limitPlaceColor = [UIColor redColor];
    textView.placeholdFont = [UIFont systemFontOfSize:17];
    textView.limitPlaceFont = [UIFont systemFontOfSize:17];
//    textView.autoHeight = @1;
    textView.limitLines = @2;//行数限制优先级低于字数限制
    textView.infoBlock = ^(NSString *text, CGSize textViewSize) {
        NSLog(@"当前文字: %@   当前高度:%lf",text,textViewSize.height);
    };
    
    [self.view addSubview:textView];
    
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: YLTextViewDemo/YLTextView_SwitDemo/YLTextView_Swit/ViewController.swift
//
//  ViewController.swift
//  YLTextView_Swit
//
//  Created by 张雨露 on 2017/6/9.
//  Copyright © 2017年 张雨露. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        view.backgroundColor = .lightGray
        let textview = UITextView(frame: CGRect(x: 100, y: 100, width: 200, height: 150))
//        textview.text = "如果你想对textView.text直接赋值。请在设置属性之前进行，否则影响计算"
        textview.placeholder = "喜欢请Star"
        textview.limitLength = 20
//        textview.limitLines = 4;
        textview.center = self.view.center
        view.addSubview(textview)
    }

}

